src/plugins/dataflow/components/dataflow-program.tsx

```diff
@@ -514,10 +522,24 @@ export class DataflowProgram extends BaseComponent<IProps, IState> {
   };
 
   private tick = () => {
-    // Update the sampling rate
     const now = Date.now();
     this.setState({lastIntervalDuration: now - this.lastIntervalTime});
     this.lastIntervalTime = now;
+    const isRecording = this.props.programRecordState === 1;
+
+    /* ==[ Per tick - create a case and write it to the dataSet ] == */
+    if (isRecording){
+      const aCase = {
+        __id__: newCaseId(),
+      };
```

Comment:
```suggestion
      const aCase: ICaseCreation = {};
```
You don't need the `__id__` because `addCanonicalCasesToDataSet()` will add it automatically. You _do_ want the `ICaseCreation` type here because it obviates the need for the cast in line 538 below.